src/std/misc/string.ss

```diff
@@ -77,3 +79,80 @@ package: std/misc
     ((_ eol fallback) (let ((trimmed (string-trim-suffix eol string)))
                         (if (eq? trimmed string) fallback (values trimmed eol)))))
   (try +crlf+ (try +lf+ (try +cr+ (values string "")))))
+
+
+;; Returns a single character-string of s at the index i
+;; or an error if the index could not be found.
+(def (string-at s i)
+  (string (string-ref s i)))
+
+
+;; string-replace helper which handels the case
+;; that the argument 'old' is an empty string.
+;;   new    non-empty
+;;   limit  non-zero
+(def (sr-helper-empty-old str new limit)
+  (def len-str (string-length str))
+  (if (= limit 1)
+    (string-append new str)
+    (let ((stop (1- (string-length str)))
+	  (limit (if (or (< limit 0)
+			 (> limit len-str)) (1+ len-str) limit)))
+      (string-concatenate
+       (cons new
+	(let loop ((i 0) (count 1))
+	  (cond
+	   ((= count limit)
+	    (cons new (cons (substring/shared str i) [])))
+	   ((= i stop)
+	    (cons (if (zero? i) "" new)
+		  (cons (string-at str i) (cons new []))))
+	   (else
+	    (cons (if (zero? i) "" new)
+		  (cons (string-at str i)
+			(loop (1+ i) (1+ count))))))))))))
+
+
+;; string-replace helper which handels the case
+;; that the argument 'old' is a non-empty string.
+;; str    non-empty
+;; old    non-empty
+;; new    can be empty
+;; limit  non-zero
+(def (sr-helper-nonempty-old str old new limit)
```

Comment:
add a `(declare (fixnum))` here, all the arithmetic operations are on fixnums.

Reply:
By luck I discovered the `fx...` (`fx1+`) procedures and used them on an updated code version. Does `(declare (fixnum))` anything better, or can I leave my new code as it is?

Reply:
It replaces all arithmetic operations in the scope with their fixnum equivalents.
So you don't have to use the explicit `fx` forms and you can leave your code as is with the declaration.

Reply:
also of note it's more aesthetically pleasing to do it with the declaration.